src/main/java/frc/robot/commands/DriveDistance2.java

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import frc.robot.subsystems.Drive;
+import frc.robot.Constants;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+/** An example command that uses an example subsystem. */
+public class DriveDistance2 extends CommandBase {
+  private final Drive m_subsystem;
+  private double target_distance = 0.0; // desired distance to travel in inches
+  private double error_inches = 2.0; // default error margin in inches
+  private double target_rotations = 0.0; // in sparkmax rotations
+  private double current_rotations = 0.0; // in sparkmax rotations
+  private double error_rotations; //in sparkmax rotations
+
+  // constructor that takes in a distance in inches and uses default error margin
+  public DriveDistance2(Drive subsystem, double distance) {
+    m_subsystem = subsystem;
+    addRequirements(subsystem);
+    target_distance = distance;
+  }
+
+  // constructor that takes in a distance in inches and specified error margin in inches
+  public DriveDistance2(Drive subsystem, double distance, double error) {
+    m_subsystem = subsystem;
+    addRequirements(subsystem);
+    target_distance = distance;
+    if(error < 1.0){
+      error_inches = 1.0;
+    }else{
+      error_inches = error;
+    }
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    m_subsystem.resetEncoders();
+    current_rotations = 0.0;
+    // convert target distance in inches into sparkmax rotations
+    target_rotations = (inchesToRotations(target_distance));
+    // convert error margin in inches into sparkmax rotations
+    error_rotations = (inchesToRotations(error_inches));
+
+    // should start robot in motion using sparkmax hardware PID/enconders
+    m_subsystem.setLeftRotations(target_rotations);
+    m_subsystem.setRightRotations(target_rotations);
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+  }
+
+  // Returns true when the command should end.
+  @Override
+  public boolean isFinished() {
+    // could also add a failsafe to stop after 10 seconds under all circumstances
+    current_rotations =  (m_subsystem.getLeftRotations() + m_subsystem.getRightRotations())/2.0;//takes average, probably unnecssary
```

Comment:
this actually coudl cause problems.  If your right motor had more drag and was way off (like we feared it might be), then left could have turned the bot 30 degrees, but this method would show that it is finsihed.

Checking both groups would allow the right to come back and straighten the bot.  Turning during drive and then correcting in the end obviously doesn't actually fix that problem (you'd have missed your mark), so .... thats not perfect either.

Reply:
I think example code shows taking the average. Also OK changing it to how you suggest.